invenio_drafts_resources/services/records/components/files.py

```diff
@@ -45,25 +47,25 @@ def update_draft(self, identity, data=None, record=None, errors=None):
         draft = record
         enabled = data.get("files", {}).get("enabled", True)
         default_preview = data.get("files", {}).get("default_preview")
-
-        try:
-            self.files_component.assign_files_enabled(draft, enabled)
-        except ValidationError as e:
-            errors.append({"field": "files.enabled", "messages": e.messages})
-            return  # exit early
-
-        if draft.files.enabled and not draft.files.items():
-            errors.append(
-                {
-                    "field": "files.enabled",
-                    "messages": [
-                        _(
-                            "Missing uploaded files. To disable files for "
-                            "this record please mark it as metadata-only."
-                        )
-                    ],
-                }
-            )
+        if self.service.check_permission(identity, "manage_files"):
```

Comment:
- Doing `.check_permission` vs. `.require_permission` so that this doesn't raise. Might need to add an error though so that the user knows this cannot be changed.
- Not a big fan of making another permission check (since we already did one in the service method), but I think the idea of components is that they can make additional (permission) checks.